docs/source/general/data.rst

```diff
@@ -2,11 +2,132 @@
 Data
 ####
 
+.. _data:
+
+*******************************
+Using DataModule + DataPipeline
+*******************************
+
```

Comment:
This is missing context.

What are these objects? Why do they exist? when do I need to use them?

I think we can do something like
1. How to use out-of-the-box flashdatamodules
2. How to customize existing datamodules
3. How to build a datamodule for a new task
4. What are data pipelines and when are they required
5. how to use them

Reply:
Yes. Definitely. I will work on this next week :)